posts/elixir-graphql-part-one.md

```diff
@@ -0,0 +1,311 @@
+
+Well, lots of things happen. Especially if you have a mobile or client application that is reliant on your data! The shape of what you might fetch for a desktop client is likely going to look radically different from what you would fetch for a mobile client, especially on a slower connection.
+
+GraphQL recognizes two things:
+
+1. The shape of your data will change
+2. The shape of your data needs to be different for each client
+
+Based on that, GraphQL implements more of a client-focused method of getting data out of a system. The client will tell the server what data it needs, and more importantly, HOW it needs that data to be shaped! This allows you to iterate a little faster, to build better filters and functionality, and easily add support for more data in your system without having to build/version a whole brand new endpoint that you will also need to support over time.
+
+Another awesome feature is that GraphQL supports querying for schema information, which means your users can do more to figure out the shape of the data and what they can do with it, which is incredibly helpful for both the client AND the server!
+
+GraphQL is not perfect, by any means. It's pretty complicated and you'll likely run into a lot of people with lots of REST experience not understanding why a particular piece of a GraphQL query works the way it does. You'll need to train your users to be able to use this new API effectively, but the end result is allowing the client to get whatever they want, however they want it!
+
+With that past us, let's move on to talk more about the actual implementation!
+
+#### Designing the shape of our data
+
+Our application is a simple event logging platform, so we're just going to have a very simple table structure (to start).
+
+    events
+    ___
+    id
+    event_type
+    message
+    payload
+    inserted_at
+    updated_at
+    
+`id`, `inserted_at`, and `updated_at` will all be included as default columns for any Ecto tables we create, so we don't need to worry about how we're going to define those.
+
+For the `event_type`, `payload`, and `message` columns, we're going to use those to track what types of events (unsurprisingly) each event is and message will store a condensed version of the payload. Separately, we have another column called `payload`, whose responsibility is to store the full payload when necessary. This allows the `message` column to stay smaller and easier to search!
+
+#### Getting our database tables running in Ecto
+
+Let's get started by building out this table using Phoenix's built-in generators:
+
+```
+$ mix phx.gen.context Log Events events event_type:string message:string payload:text
```

Comment:
Any particular reason you're namespacing with `Log` here? It appears the files generated are not consistent. Ex: `events.ex` and then `log/log.ex` / `log/log_test.exs`. I think we should just stick with `LogEvent` (one word).